resources/replicaset.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2016 Mirantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"errors"
+	"fmt"
+	"log"
+
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+type ReplicaSet struct {
+	ReplicaSet    *extensions.ReplicaSet
+	Client        unversioned.ReplicaSetInterface
+	SuccessFactor int32
+}
+
+func replicaSetStatus(r unversioned.ReplicaSetInterface, name string, successFactor int32) (string, error) {
+	rs, err := r.Get(name)
+	if err != nil {
+		return "error", err
+	}
+
+	if rs.Status.Replicas < rs.Spec.Replicas * 100 / successFactor {
+		return "not ready", nil
+	}
+
+	return "ready", nil
+}
+
+func replicaSetKey(name string, successFactor int32) string {
+	return fmt.Sprintf("replicaset/%s,%d", name, successFactor)
```

Comment:
I'm not sure that we need successFactor as a part of Key. Pls note, that that is thew Key, which user should provide in dependencies. That could also confuse user since one could think we could have separate dependencies on the same RC but with different successFactor, but that is not true.